astacus/coordinator/list.py

```diff
@@ -39,6 +41,22 @@ def compute_deduplicated_snapshot_file_stats(manifest: ipc.BackupManifest) -> tu
     return num_files, total_size
 
 
+def get_object_storage_size_bytes(manifest: ipc.BackupManifest) -> int | None:
+    """Compute the total size of object storage files for the backup in bytes."""
+    if manifest.plugin == ipc.Plugin.clickhouse:
+        clickhouse_data = ClickHouseManifest.from_plugin_data(manifest.plugin_data)
```

Comment:
isn't it expensive to parse?... would it be better to add a new field to the base manifest, set to 0 for all other service types and populate it for CH when creating the backup? e.g. add something a-la:

```
class BackupManifest(...
...
    # What did the upload return (mostly for statistics)
    upload_results: Sequence[SnapshotUploadResult]

    # How much bytes are we keeping in object storage
    tiered_storage_results: Sequence[TieredStorageResult]
```
to ipc.BackupManifest? Recalculating this value feels wrong, because manifests are supposed to be immutable.

Reply:
Yea agreed that parsing the plugin data is not great.  Although I have been thinking about [this](https://jcristharif.com/msgspec/supported-types.html#generic-types) to make it faster potentially.

Since `UploadManifestStep` is shared, what is the least awkward way to add this to the base manifest do you think?

Reply:
BTW confusingly `clickhouse_data.object_storage_files` will usually have length 1 (since it has one value per CH disk) so it is not doing much computation after reparsing the plugin data.